### PR TITLE
Bug 1970828: fix projected service accounts as oauth-client

### PR DESCRIPTION
**Pull request by stlaz**
- [x] unit tests will fail (fixed)
- [x] contains unsafe debug logs (fixed)

### CODE REVIEW

**Comment by stlaz** [comment]
/assign @deads2k 
cc @s-urbaniak 

this could work, the work with audiences feels a little wobbly

**Comment by s-urbaniak** [comment]
/retitle Bug 1970828: fix projected service accounts as oauth-client

File: pkg/osinserver/registrystorage/storage.go [hunk 1]
```diff
@@ -70,6 +77,32 @@ func (w *clientWrapper) ClientSecretMatches(secret string) bool {
...
+		if err != nil {
+			klog.V(4).Infof("the tokenreview failed: %v", err)
+			return false
+		}
+
+		if tokenReview.Status.User.Username != w.client.Name {
+			return false
+		}
+		return true
+	}
```
Comment (s-urbaniak) [inline pkg/osinserver/registrystorage/storage.go:104]: follow-up nit:
```
if tok == nil {
  return false
}

public := &jwt.Claims{}
...
```

Comment (s-urbaniak) [reply pkg/osinserver/registrystorage/storage.go:104]: (unindent the if-block)

**Comment by s-urbaniak** [comment]
/test e2e-agnostic-upgrade
e2e failure is unrelated

**Pull request merged**